e2e/tests/onchain_settlement.rs

```diff
@@ -169,7 +169,7 @@ async fn onchain_settlement(web3: Web3) {
         liquidity_collector,
         price_estimator,
         vec![Box::new(solver)],
-        Box::new(web3.clone()),
+        Arc::new(web3.clone()),
```

Comment:
Because we want access to the "same" price estimator in multiple components. Box is an exclusive pointer to an object (not directly cloneable), whereas Arc is a shared pointer to the underlying object which can be cloned.